terraformignore.go

```diff
@@ -0,0 +1,219 @@
+package slug
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"text/scanner"
+)
+
+func parseIgnoreFile(rootPath string) []rule {
+	// Look for .terraformignore at our root path/src
+	file, err := os.Open(filepath.Join(rootPath, ".terraformignore"))
+	defer file.Close()
+
+	// If there's any kind of file error, punt and use the default ignore patterns
+	if err != nil {
+		// Only show the error debug if an error *other* than IsNotExist
+		if !os.IsNotExist(err) {
+			fmt.Printf("Error reading .terraformignore, default exclusions will apply: %v \n", err)
```

Comment:
I think errors should be going to stderr in general, but I'm not sure where the output of this fits into the Terraform UI.

Reply:
I very much doubt if a package like this should write anything to either `stdout` or `stderr`. Wouldn't it be cleaner to return any errors or warnings and let the caller decide what to do with them?

Reply:
Good point, I'll update it at least to go to standard error.

I'm also a bit torn on what to do about Terraform UI -- since this is consumed by go-tfe, the function in go-tfe would need to update to handle errors, and I was attempting to avoid needing to change the function signature in this round, but if _that_ handles the error, it can then pass that up again through to the remote backend, at which point we have tfdiags at our disposal to show things the "right" way.

Reply:
Practically, yes, that would be correct. However, the caller in this case will be the `Pack` function, which would then have this same decision, and upstream go-tfe currently bails on any error case.